app/src/main/java/com/example/calculator/Calculator.kt

```diff
@@ -0,0 +1,208 @@
+    printHelp()
+    while (true) {
+        val line = readLine()
+        when {
+            (line?.toString() == "help") -> printHelp()
+            (line?.toString() == "quit") -> return
+            else -> {
+                when (val result = proceedOrdinaryOperations(calculator, line) ) {
+                    is CommandResult.Error -> {
+                        println(result.message)
+                    }
+                    is CommandResult.Success -> {
+                        val result = calculator.getResult()
+                        if ( result.toDouble() == 0.0) {
+                            println(0)
+                        } else {
+                            println(result)
+                        }
+                    }
+                }
+//                val indexOfFirstOpenBracket = line?.indexOfFirst { it.toString() == "(" }
+//                val indexOfFirstCloseBracket = line?.indexOfFirst { it.toString() == ")" }
+//                if (indexOfFirstOpenBracket != null) {
+//                    when {
+//                        indexOfFirstCloseBracket == -1 && indexOfFirstOpenBracket == -1 -> {
+//                            when (val result = proceedOrdinaryOperations(calculator, line) ) {
+//                                is CommandResult.Error -> {
+//                                    println(result.message)
+//                                    return
+//                                }
+//                                is CommandResult.Success -> {
+//                                    println(calculator.getResult())
+//                                }
+//                            }
+//                        }
+//                        (indexOfFirstOpenBracket == -1 && indexOfFirstCloseBracket != -1) ||
+//                                (indexOfFirstOpenBracket != -1 && indexOfFirstCloseBracket == -1) -> {
+//                            println(CommandResult.WRONG_INPUT)
+//                        }
+//                        indexOfFirstOpenBracket > indexOfFirstCloseBracket!! -> {
+//                            println(CommandResult.WRONG_INPUT)
+//                        }
+//                        else -> {
+//                            if (line?.count { it.toString() == "(" } != line?.count { it.toString() == ")" }) {
+//                                println(CommandResult.WRONG_INPUT)
+//                            } else {
+//                                proceedOperationsWithBrackets(calculator, line)
+//                            }
+//                        }
+//                    }
+//                }
+            }
+        }
+    }
+}
+
+fun proceedOperationsWithBrackets(calculator: CommandProcessor, line: String?) {
+    val listOfNestedStrings = mutableListOf<String>()
+    recursiveSplit(calculator,line,listOfNestedStrings, 0)
+}
+
+fun recursiveSplit(calculator: CommandProcessor, line: String?, listOfNestedStrings: MutableList<String>, startedIndex: Int) {
```

Comment:
лучше было бы так, чтобы избежать мутабельных объектов
```suggestion
fun recursiveSplit(calculator: CommandProcessor, line: String?, startedIndex: Int): List<String> {
```